.github/workflows/integration.yml

```diff
@@ -0,0 +1,95 @@
+name: Integration
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  kind:
+    name: Kind
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Go
+        id: setup-go
+        uses: ./.github/actions/setup-go-cache
+        with:
+          cache-prefix: go-integration-kind
+
+      - name: Starting Armada cluster
+        run: make kind-all
+
+      - name: Create queue
```

Comment:
I am more in favour of an approach where the test is defined in a bash script, as we can also execute it locally.

Reply:
I agree, but all this boils down to:
```
./bin/app/armadactl create queue example
./bin/app/armadactl submit dev/quickstart/example-job.yaml
```
which is taken from the `README.md`.

The boilerplate code is hopefully only temporary until #321 is fixed.

Reply:
I don't think https://github.com/armadaproject/armada-operator/issues/321 will be fixed, as Armada is eventually consistent by design, and clients should handle retries.

Reply:
would be nice if the `armadactl` would have an option to wait for the consistency, then users wouldn't need to implement that individually

Reply:
Might be worthwhile opening an Issue in the https://github.com/armadaproject/armada repository.

Reply:
I have found `scripts/e2e-test.sh`, moving code there and invoking this from CI.